### ISSUE TITLE
Unspecified modification written to DDA-produced library file

### ISSUE THREAD

**Issue by MassSpecPostDoc** [Nesvilab/FragPipe#1051]
Hello,

The workflow for "Spectral library building from DDA (SpecLib)" was followed and appeared to be successful. Some specific variable mass shifts were added to the MSFragger settings prior to running (see image).

The resulting combined_modified_peptide.tsv file indicated the specified PTMs were present, including the modified N-terminal methionine with the specified mass shift (114), and the modified lysine with the same mass shift (114). Note that the PTM in question is UniMod:121, i.e. this is a reasonably well studied modification, and is known to occur predominantly on lysine but can occur on protein N-termini, so therefore often methionine.

The problem appears in the resulting library.tsv file, where the ModifiedPeptideSequence column has the modification on lysine annotated as (UniMod:121) which is fine, but it looks like the N-terminal methionine modification has been annotated as a different mass shift, i.e. not 114, instead it is shown as M[245.08344]. The stripped peptide sequences match those shown as possessing M[114.0429] in the combined_modified_peptide.tsv file. None of the identified N-terminal methionines possess the correctly annotated modification/mass, instead, they are annotated with M[245.08344]. 

I do not know where the M[245.08344] mass shift comes from, this is not anywhere in the MSFragger settings, nor can I find any mass shifts starting with 245 in the UniMod database. the mass shift is almost the same as an oxidised methionine and 2 x UiMod:121, but this is not likely as it should not occur, and the mass shift does not quite add up in this scenario. 

To sum up: In the combined_modified_peptide.tsv file, peptides seem to be annotated with the correct Mass shift, why is the mass shift incorrect in the library.tsv file? Am I doing something wrong? 

Thank you very much for your Attention,

Andy

![image](https://user-images.githubusercontent.com/101635843/227500595-93111ec3-c5a4-489b-ab1a-197c07a08348.png)

**Comment by MassSpecPostDoc** [comment]
Hello Alexey,

OK that's great. DIA-NN actually failed using that library file,  it said something about the modification being unspecified. I will either use a replace function to swap the 245 annotation for UniMod:121, or just remove it from the library since the lysine mod is more important anyway. Thank you for letting me know.

Cheers,

Andy

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by anesvi** [comment]
I think DIA-NN failed because in DIA-NN tab we now have 'replace library spectra with predicted' option checked by default.  Please uncheck that are rerun (just DIA-NN, uncheck all other tabs) and it should work

**Comment by fcyu** [comment]
In the future, we need to add `--strip-unknown-mods` to DIA-NN when `replace library spectra with predicted` is enabled.

Best,

Fengchao

**Comment by fcyu** [comment]
Fixed.
